### PR TITLE
libsel4arm-vmm: fix bugs in ba8de51828024fe1de88b2

### PR DESCRIPTION
**Pull request by pingerino**
- out-by-1 in LR_OF_NEXT
  - bad assertion in vgic_vcpu_inject_irq
  - inifinite loop in handle_vgic_maintenance

  handle_vgic_maintenance now processes as many overflow irqs as existed
  at the start of the procedure call.

### CODE REVIEW

File: libsel4arm-vmm/src/arch/arm/devices/vgic.c [hunk 1]
```diff
@@ -297,11 +297,11 @@ static int vgic_vcpu_inject_irq(struct device *d, vm_t *vm, struct virq_handle *
         return err;
     } else {
         /* Add to overflow list */
-        int idx = LR_OF_NEXT(vgic->lr_overflow.tail);
-        ZF_LOGF_IF(idx == vgic->lr_overflow.head && idx != vgic->lr_overflow.tail,
-                   "too many overflow irqs");
+        int idx = vgic->lr_overflow.tail;
+        ZF_LOGF_IF(idx == vgic->lr_overflow.head &&
+                   vgic->lr_overflow.head != vgic->lr_overflow.tail, "too many overflow irqs");
```
Comment (WellMcGarnicle) [inline libsel4arm-vmm/src/arch/arm/devices/vgic.c]: Why is it an error if the head does not equal the tail? Also if the first condition is true the second must be false?

Comment (pingerino) [reply libsel4arm-vmm/src/arch/arm/devices/vgic.c]: When `head == tail` the buffer is empty, so we can allocated the head in this case.

Comment (pingerino) [reply libsel4arm-vmm/src/arch/arm/devices/vgic.c]: otherwise the buffer is full and we're about to overwrite the current head

Comment (pingerino) [reply libsel4arm-vmm/src/arch/arm/devices/vgic.c]: Thanks for the fix!

**Pull request merged**